### PR TITLE
Block Sum of Matrix in C++

### PR DESCRIPTION
**Pull request by U-c0de**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
(YES)

### Description
Block Sum of Matrix (Submatrix block addition according to coordinates) | C++ | Dynamic Programming

### Checklist
- [ yes] I've read the contribution guidelines.
- [ yes] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

 #6425

### CODE REVIEW

**Comment by U-c0de** [comment]
Please Check -- @HarshCasper @gat786

**Comment by U-c0de** [comment]
@ankitaggarwal23  Do check !!

**Comment by U-c0de** [comment]
Please check this @siddharth25pandey

File: C-Plus-Plus/dp/BlockSumMatrix.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                  }
+               }
+            }
+         }
+      }
+      return res;
+   }
+};
+main(){
+   vector<vector<int>> sz1 = [[1,2,3], [4,5,6], [7,8,9]];  //put the input from test cases
```
Comment (siddharth25pandey) [inline C-Plus-Plus/dp/BlockSumMatrix.cpp]: Take Dynamic input from the user with displaying  message in print form, as what are you taking input from the user

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@U-c0de Small changes are required

**Comment by U-c0de** [comment]
@siddharth25pandey  Now check

**Comment by U-c0de** [comment]
Changes made @siddharth25pandey

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.tutorialspoint.com/matrix-block-sum-in-cplusplus)

**Pull request closed without merging**